source/extensions/filters/http/transformation/transformer.h

```diff
@@ -49,14 +53,52 @@ class Transformer {
 typedef std::shared_ptr<Transformer> TransformerSharedPtr;
 typedef std::shared_ptr<const Transformer> TransformerConstSharedPtr;
 
+class TransformerPair {
+public:
+  TransformerPair(TransformerConstSharedPtr request_transformer, TransformerConstSharedPtr response_transformer):
+    request_transformation_(request_transformer), response_transformation_(response_transformer) {}
+
+  TransformerConstSharedPtr getRequestTranformation() const {
+    return request_transformation_;
+  }
+
+  TransformerConstSharedPtr getResponseTranformation() const {
+    return response_transformation_;
+  }
+  
+private:
+  TransformerConstSharedPtr request_transformation_;
+  TransformerConstSharedPtr response_transformation_;
+};
+
+typedef std::shared_ptr<const TransformerPair> TransformerPairConstSharedPtr;
+
+class MatcherTransformerPair {
+public:
+  MatcherTransformerPair(Matcher::MatcherConstPtr matcher, TransformerPairConstSharedPtr transformer_pair)
+      : matcher_(matcher), transformer_pair_(transformer_pair) {}
+
+  TransformerPairConstSharedPtr transformer_pair() const {
+    return transformer_pair_;
+  }
+
+  Matcher::MatcherConstPtr matcher() const {
+    return matcher_;
+  }
+
+private:
+  Matcher::MatcherConstPtr matcher_;
+  TransformerPairConstSharedPtr transformer_pair_;
+};
+
+
 class TransormConfig {
 public:
   virtual ~TransormConfig() {}
+
+  virtual TransformerPairConstSharedPtr findTransformers(const Http::HeaderMap& headers) const PURE;
   
-  virtual TransformerConstSharedPtr getRequestTranformation() const PURE;
   virtual bool shouldClearCache() const PURE;
```

Comment:
if TransformerPairConstSharedPtr goes into TransformerPair it can be removed from here

Reply:
What do you mean by this?